### PR TITLE
flaca: init at 2.2.1

### PR DESCRIPTION
**Pull request by zzzsyyy**
###### Description of changes

[Flaca](https://github.com/Blobfolio/flaca) is a CLI tool for x86-64 Linux machines that simplifies the task of maximally, losslessly compressing JPEG and PNG images for use in production **web environments**.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/flaca/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchFromGitHub {
+    owner = "Blobfolio";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-RXMqPpQM2h6EtIIH8Ix31euC7zK3v2QohZqouNlK7rM=";
+  };
+
+  cargoPatches = [ ./cargo-lock.patch ];
+  cargoHash = "sha256-z8nqpaSz5dZuoDJUgX1n+6gwFYsS42xwgCn7RnzBXbw=";
```
Comment (emilylange) [inline pkgs/applications/graphics/flaca/default.nix:15]: Please use the following instead :)

```suggestion
  # upstream does not provide a Cargo.lock
  cargoLock.lockFile = ./Cargo.lock;
  postPatch = ''
    ln -s ${./Cargo.lock} Cargo.lock
  '';
```

And replace `./cargo-lock.patch` with the actual `Cargo.lock` (plain, not as `diff`)

File: maintainers/maintainer-list.nix
Comment (emilylange) [inline maintainers/maintainer-list.nix:1]: Please split this change to `maintainers/maintainer-list.nix` into to a separate commit called
`maintainers: add zzzsy`

See https://nixos.org/manual/nixpkgs/unstable/#var-meta-maintainers

So two commits in total:
1. `maintainers: add zzzsy`
1. `flaca: init at 2.2.1`

Thank you!

File: pkgs/applications/graphics/flaca/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+  meta = with lib; {
+    description = "A CLI tool for x86-64 Linux machines that simplifies the task of maximally, losslessly compressing JPEG and PNG images for use in production web environments";
```
Comment (emilylange) [inline pkgs/applications/graphics/flaca/default.nix:18]: IMHO a bit long for a short description.

How about

```suggestion
    description = "A CLI tool to losslessly compress JPEG and PNG images";
```

You can put your original description into `longDescription`, if you like :)

https://nixos.org/manual/nixpkgs/unstable/#var-meta-description

**Review by emilylange — CHANGES_REQUESTED** [review]

File: pkgs/applications/graphics/flaca/default.nix (hunk 1 above)
Comment (zzzsyyy) [reply pkgs/applications/graphics/flaca/default.nix]: thank you for your suggestion!

File: pkgs/applications/graphics/flaca/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+  # upstream does not provide a Cargo.lock
+  cargoLock.lockFile = ./Cargo.lock;
+  postPatch = ''
+    ln -s ${./Cargo.lock} Cargo.lock
+  '';
+
+  meta = with lib; {
+    description = "A CLI tool to losslessly compress JPEG and PNG images";
+    longDescription = "A CLI tool for x86-64 Linux machines that simplifies the task of maximally, losslessly compressing JPEG and PNG images for use in production web environments";
```
Comment (emilylange) [inline pkgs/applications/graphics/flaca/default.nix:22]: I am honestly a bit confused why upstream is so focused on the `x86-64 Linux` wording.
I've tested it on `aarch64-linux` with a random `.png` and it seems to work great.

Ehh whatever. Not suggesting a change. Just wanted to note that.

**Review by emilylange — APPROVED** [review]
Thank you!

**Pull request merged**